### PR TITLE
Add examples to fit function docstrings

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This is to add examples to various fit functions, as mention in comment here 
https://github.com/gammapy/gammapy/issues/5291#issuecomment-2163318629

### CODE REVIEW

File: gammapy/modeling/fit.py [hunk 1]
```diff
@@ -513,6 +559,24 @@ def stat_contour(self, datasets, x, y, numpoints=10, sigma=1):
...
+        ...     dataset = SpectrumDatasetOnOff.read(
+        ...         f"$GAMMAPY_DATA/joint-crab/spectra/hess/pha_obs{obs_id}.fits"
+        ...     )
+        ...     datasets.append(dataset)
+        >>> datasets = datasets.stack_reduce(name="HESS")
+        >>> model = SkyModel(spectral_model=LogParabolaSpectralModel(), name="crab")
+        >>> datasets.models = model
+        >>> fit = Fit(backend='minuit')
+        >>> optimize = fit.optimize(datasets)
+        >>> stat_contour = fit.stat_contour(datasets=datasets, x="alpha", y="amplitude")
```
Comment (AtreyeeS) [inline gammapy/modeling/fit.py:579]: having `x` and `y` as `parameter` directly is probably clearer

File: gammapy/modeling/fit.py [hunk 2]
```diff
@@ -442,6 +461,33 @@ def stat_surface(self, datasets, x, y, reoptimize=False):
...
+        ...         f"$GAMMAPY_DATA/joint-crab/spectra/hess/pha_obs{obs_id}.fits"
+        ...     )
+        ...     datasets.append(dataset)
+        >>> datasets = datasets.stack_reduce(name="HESS")
+        >>> model = SkyModel(spectral_model=LogParabolaSpectralModel(), name="crab")
+        >>> datasets.models = model
+        >>> par_alpha = datasets.models.parameters["alpha"]
+        >>> par_beta = datasets.models.parameters["beta"]
+        >>> par_alpha.scan_values = np.linspace(1.55, 2.7, 20)
+        >>> par_beta.scan_values = np.linspace(-0.05, 0.55, 20)
```
Comment (AtreyeeS) [inline gammapy/modeling/fit.py:483]: If this takes too long to run, you can decrease the number of points

**Review by AtreyeeS — DISMISSED** [review]
Thank you @Astro-Kirsty ! I have one small inline comment about directly passing the parameter instead of via names

File: gammapy/modeling/fit.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/fit.py:483]: tried it, it's quite fast

File: gammapy/modeling/fit.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/fit.py]: done

**Review by QRemy — APPROVED** [review]
Thanks @Astro-Kirsty,  no further comment.

**Review by AtreyeeS — APPROVED** [review]
Thanks @Astro-Kirsty and @QRemy ! Merging this now

**Pull request merged**